### ISSUE TITLE
Feature - Auth - 2 Factor Authentication

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/TalentCloud#1570]
# Description

Implement 2 Factor Authentication for login. It must be required for Validated Managers (at least when they view applicants/applications) and Admins, but optional for other users.

More details waiting on spike #1569. 

- [ ] This task is complex and needs acceptance criteria.

**Comment by tristan-orourke** [comment]
Closing this issue because we've split it into multiple stories under epic #1670.
